doc/sphinxman/source/scf.rst

```diff
@@ -816,23 +816,72 @@ To avoid this, either set |scf__df_basis_scf| to an auxiliary
 basis set defined for all atoms in the system, or set |scf__df_scf_guess|
 to false, which disables this acceleration entirely.
 
+.. _`sec:directscf`:
+
+Integral-Direct Self-Consistent Field
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The integral-direct SCF algorithm [Almloef:1982:385]_ computes the
+exact Fock matrix by recomputing integrals on the fly at every
+iteration and contracting them with the density matrix. As the
+electron density changes very little in later iterations, the direct
+SCF method often employs incremental formation for the Fock matrix:
+since the Fock matrix is linear in the density, one does not have to
+compute the full matrix at every iteration, only the change from a
+previously computed Fock matrix.  This behavior is toggled by the
+|scf__incfock| keyword. Because the elements in the difference density
+matrix are much smaller than the elements in the density matrix, this
+makes the screening of the integrals much tighter when density-based screening is used, reducing the number
+of integrals that need to be calculated.
+
+The typical implementation of incremental Fock matrix formation
+updates the reference at every iteration. This makes the screening
+better and better as the wave function converges, because the density
+changes between SCF iterations go to zero as convergence is approached. But, since small changes can accumulate to a
+non-negligible outcome over many iterations, this procedure is
+susceptible to creeping numerical error. For this reason, a fresh Fock
+matrix needs to be rebuilt every ``N`` iterations
+(|scf__incfock_full_fock_every|, defaults to 5). In addition, due to the
+danger of the creepup of small errors, the iterative formation is
+turned off when a sufficiently tight convergence
+(|scf__incfock_convergence|, defaults to ``1e-5``) has been reached,
+so that the final iterations are performed without deleterious
+screening.
+
+Psi4 also implements a different type of algorithm, where a fixed
+reference density and Fock matrix is employed (see
+[Parrish:2016:131101]_ for inspiration). Because the reference is
+fixed, there is no danger of incremental creepup of numerical noise;
+however, the flip side is that roughly the same number of integrals
+need to be computed for every incremental step. This algorithm is used
+instead of the typical algorithm discussed above, if
+|scf__incfock_fixed_reference| is ``true``.
+
+
+It has the following options
+
+  |scf__incfock|: Toggles incremental formation of the Fock matrix. Defaults to ``true``.
+  |scf__incfock_fixed_reference|: Chooses the algorithm: use a fixed reference Fock matrix instead of updating it every iteration? Defaults to ``false``.
+  |scf__incfock_full_fock_every|: The number of iterations between full rebuilds of the Fock matrix. Defaults to ``5``.
+  |scf__incfock_convergence|: The convergence threshold to turn off incremental formation of the Fock matrix. Defaults to ``1e-5``.
```

Comment:
```suggestion
.. include:: autodir_options_c/scf__incfock.rst
.. include:: autodir_options_c/scf__incfock_fixed_reference.rst
.. include:: autodir_options_c/scf__incfock_full_fock_every.rst
.. include:: autodir_options_c/scf__incfock_convergence.rst
```
*If* all the description and default is in read_options.cc, you can let this include it all for you.